docs/admin/authentication.md

```diff
@@ -60,19 +59,213 @@ more secure modes described above easier to use.
 The basic auth file format is implemented in `plugin/pkg/auth/authenticator/password/passwordfile/...`
 and is a csv file with 3 columns: password, user name, user id.
 
+```conf
+password,user,uid
+```
+
 When using basic authentication from an http client, the apiserver expects an `Authorization` header
 with a value of `Basic BASE64ENCODED(USER:PASSWORD)`.
 
-**Keystone authentication** is enabled by passing the `--experimental-keystone-url=<AuthURL>`
+## Service Account Tokens
+
```

Comment:
Put this somewhere, maybe at the top:  

All Kubernetes clusters have two categories of users.  

1. Service Accounts are accounts that are managed by Kubernetes:
  - some are automatically created, but they can also be manually created
  - they are associated with a single Kubernetes cluster and namespace
  - they are automatically deleted when a namespace is deleted
  - credentials are automatically generated for service accounts
  - credentials are automatically mounted in volumes for Pods. 
  - They are intended to be used by programs running in Pods.
  - Service account names are scoped to namespaces.
2. Normal users are not managed by Kubernetes
  - The canonical storage for normal users is assumed to be outside Kubernetes.
  - The lifecycle for normal users is assumed to be independent of Kubernetes clusters and objects.
  - Normal user names are not scoped to a cluster or namespace.
  - Kubernetes does not have objects which represent normal user accounts.
  - Kubernetes authenticators identify the user who is making a request.